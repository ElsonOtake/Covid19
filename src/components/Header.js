const Header = () => (
  <header>
    <div>
      <i className="fa-solid fa-earth-americas" />
    </div>
    <p>Covid19 in South America</p>
    <div>
      <i className="fa-solid fa-microphone" />
      <i className="fa-solid fa-gear" />
    </div>
  </header>
);

export default Header;
