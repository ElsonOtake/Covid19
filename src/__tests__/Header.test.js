import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';

describe('Tests for the Header component', () => {
  render(
    <Router>
      <Header />
    </Router>,
  );
  test('Check for the following test on screen', () => {
    expect(screen.getByText(/America/)).toBeInTheDocument();
    expect(screen.getByText('Covid19 in South America')).toBeInTheDocument();

    expect(screen.getByRole('banner')).toBeInTheDocument();
    expect(screen.getByRole('banner')).toBeVisible();
    expect(screen.getByRole('link')).toBeInTheDocument();
    expect(screen.getByRole('link')).toBeVisible();
  });
  test('Check for the snapshot', () => {
    expect(screen.debug()).toMatchSnapshot();
  });
});
