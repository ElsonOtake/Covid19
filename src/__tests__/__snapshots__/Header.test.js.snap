// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Header component Check for the snapshot 1`] = `undefined`;
